NumPy 2 compatibility

IOData (or some of its unit tests) are not compatible with NumPy 2.0, which was released recently. Some tests fail, as can be seen in #339 

At first sight, all test failures are due to the same problem:

```
>       return np.array(array, copy=False, dtype=dtype)
E       ValueError: Unable to avoid copy while creating an array as requested.
E       If using `np.array(obj, copy=False)` replace it with `np.asarray(obj)` to allow a copy when needed (no behavior change in NumPy 1.x).
E       For more details, see https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword.
```

This should be easy to fix.

--- a/iodata/attrutils.py
+++ b/iodata/attrutils.py
@@ -31,7 +31,7 @@ def convert_array_to(dtype):
     def converter(array):
         if array is None:
             return None
-        return np.array(array, copy=False, dtype=dtype)
+        return np.asarray(array, dtype=dtype)
 
     return converter
 

